### PR TITLE
Refactor/loads2

### PR DESCRIPTION
**Pull request by devgabrielcoman**
A part of on-going set of PRs that basically split the refactor process of the UI SDK, now that we have Protobufs & Data SDK done to a satisfactory level.

In this PR I've tried to move some of the logic of ViewModels into ViewState objects that are immutable as well as started refactoring FollowingView and SimpleFeedView.

### CODE REVIEW

File: popjamui/src/main/java/tv/superawesome/pj/ui/viewmodels/feedcell/detail/DetailFeedCellViewModel.kt [hunk 1]
```diff
@@ -40,7 +40,7 @@ class DetailFeedCellViewModel(override var delegate: IViewModelDelegate? = null,
             field = value
 
             // notify sub view models
-            userViewModel.data = value?.postingUser
+//            userViewModel.data = value?.postingUser
```
Comment (Tbruni85) [inline popjamui/src/main/java/tv/superawesome/pj/ui/viewmodels/feedcell/detail/DetailFeedCellViewModel.kt]: Useless comment

File: popjamui/src/main/java/tv/superawesome/pj/ui/views/detailfeed/DetailFeedCell.kt [hunk 2]
```diff
@@ -24,7 +23,7 @@ class DetailFeedCell(itemView: View, val context: Context): RecyclerView.ViewHol
             field = value
 
             // assign other view models
-            itemView.UserIcon.viewModel = value?.userViewModel?.avatarViewModel
+//            itemView.UserIcon.viewModel = value?.userViewModel?.avatarViewModel
```
Comment (Tbruni85) [inline popjamui/src/main/java/tv/superawesome/pj/ui/views/detailfeed/DetailFeedCell.kt]: useless comment

File: popjamui/src/main/java/tv/superawesome/pj/ui/views/detailfeed/DetailFeedCell.kt [hunk 3]
```diff
@@ -38,10 +37,11 @@ class DetailFeedCell(itemView: View, val context: Context): RecyclerView.ViewHol
     // IViewModelDelegate
 
     override fun viewModelDidUpdate() {
```
Comment (Tbruni85) [inline popjamui/src/main/java/tv/superawesome/pj/ui/views/detailfeed/DetailFeedCell.kt]: the method doesn't do anything, I would remove it

File: popjamui/src/main/java/tv/superawesome/pj/ui/views/following/FollowingCell.kt [hunk 4]
```diff
@@ -6,36 +6,50 @@ import android.view.View
...
+import tv.superawesome.pj.ui.viewmodels.user.UserViewState
 
 /**
  * Created by gabriel.coman on 31/01/2018.
  */
-class FollowingCell(itemView: View, val context: Context): RecyclerView.ViewHolder (itemView), IView<IUserViewModel>, IViewModelDelegate {
+class FollowingCell(itemView: View, val context: Context): RecyclerView.ViewHolder (itemView), IView<IUserViewModel>, IUserViewModelDelegate, View.OnClickListener  {
+
+
+    // Init
```
Comment (Tbruni85) [inline popjamui/src/main/java/tv/superawesome/pj/ui/views/following/FollowingCell.kt]: can you use `MARK:` as in Swift to create a shortcut to easy navigate through the content of a class?

File: app/src/main/java/tv/superawesome/pj/popjamuidemo/UserActivity.kt [hunk 5]
```diff
@@ -26,33 +32,31 @@ class UserActivity: AppCompatActivity() {
...
+        val feedService = sdk.getService(type = IFeedService::class.java)!!
+
+        val userProvider = UserDataProvider(loggedUser = loggedUser, userService = userService)
+        val followProvider = FollowUnfollowDataProvider(loggedUser = loggedUser, actionService = userActionsService)
+        val userViewModel = UserViewModel(userProvider = userProvider, followUnfollowProvider = followProvider)
         userProvider.delegate = userViewModel
-        buttonProvider.delegate = userViewModel.followButtonViewModel
         UserView.viewModel = userViewModel
+        UserView.delegate = this
 //        UserView.init(withUserId = loggedUser.userId)
```
[the remark this answers is not in the document]
Comment (devgabrielcoman) [reply app/src/main/java/tv/superawesome/pj/popjamuidemo/UserActivity.kt]: agreed, but again, this is in the demo app that I used to visually-test the SDK. this is not the vanilla app

File: app/src/main/java/tv/superawesome/pj/popjamuidemo/UserActivity.kt [hunk 6]
```diff
@@ -26,33 +32,31 @@ class UserActivity: AppCompatActivity() {
         val loggedUser = session?.getCurrentUser(context = this)!!
         val userService = sdk.getService(type = IUserSocialService::class.java)!!
         val userActionsService = sdk.getService(type = IUserSocialActionsService::class.java)!!
```
[the remark this answers is not in the document]
Comment (devgabrielcoman) [reply app/src/main/java/tv/superawesome/pj/popjamuidemo/UserActivity.kt]: agreed, but again, this is in the demo app that I used to visually-test the SDK. this is not the vanilla app

File: popjamui/src/main/java/tv/superawesome/pj/ui/viewmodels/feedcell/detail/DetailFeedCellViewModel.kt (hunk 1 above)
Comment (devgabrielcoman) [reply popjamui/src/main/java/tv/superawesome/pj/ui/viewmodels/feedcell/detail/DetailFeedCellViewModel.kt]: The whole code is still in major refactoring mode :)

File: popjamui/src/main/java/tv/superawesome/pj/ui/views/following/FollowingCell.kt (hunk 4 above)
Comment (devgabrielcoman) [reply popjamui/src/main/java/tv/superawesome/pj/ui/views/following/FollowingCell.kt]: nope :(

**Pull request closed without merging**